From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jonas Karlman <jonas@kwiboo.se>
Date: Tue, 4 Aug 2026 21:44:39 +0000
Subject: rockchip: rk3506: Remove unused rockchip_get_clk() helper

The rockchip_get_clk() helper is typically used by SDRAM init code of
older SoC generations or to help get the CRU base address from the clock
driver private data.

RK3506 has no uses for the rockchip_get_clk() helper, let's drop it.

Signed-off-by: Austin Lane <vidplace7@gmail.com>
---
 arch/arm/mach-rockchip/rk3506/clk_rk3506.c | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/arch/arm/mach-rockchip/rk3506/clk_rk3506.c b/arch/arm/mach-rockchip/rk3506/clk_rk3506.c
index 111111111111..222222222222 100644
--- a/arch/arm/mach-rockchip/rk3506/clk_rk3506.c
+++ b/arch/arm/mach-rockchip/rk3506/clk_rk3506.c
@@ -1,15 +1,9 @@
 // SPDX-License-Identifier: GPL-2.0-or-later
 // Copyright Contributors to the U-Boot project.
 
-#include <dm.h>
+#include <linux/kernel.h>
 #include <asm/arch-rockchip/cru_rk3506.h>
 
-int rockchip_get_clk(struct udevice **devp)
-{
-	return uclass_get_device_by_driver(UCLASS_CLK,
-				DM_DRIVER_GET(rockchip_rk3506_cru), devp);
-}
-
 void *rockchip_get_cru(void)
 {
 	return (void *)RK3506_CRU_BASE;
-- 
Armbian

